app/controllers/bookmarks/search_controller.rb

```diff
@@ -2,18 +2,38 @@ class Bookmarks::SearchController < ApplicationController
   # GET /bookmarks/search
   # GET /bookmarks/search.json
   def index
-    @bookmarks = BookmarksIndex.query(
-      bool: {
-        should: [
-          { match: { title: params[:q] } },
-          { match: { tags: params[:q] } }
-        ]
-      }
-    ).objects.page params[:page]
+    @query = params[:q]
+    @searcher = policy_scope(BookmarkSearcher)
+
+    if @query.blank? || @query.empty?
+      @bookmarks = Bookmark.none
+    else
+      if @query
+        odd_quotes = @query.count('"').odd?
+        @query = @query + '"' if odd_quotes
+
+        @searcher.query @query
+        @searcher.errors.add :query, "Quotation marks were unbalanced, an extra quote was added to the end of the query" if odd_quotes
+      end
+
+      @bookmarks = @searcher.results
+        .includes(
+          :webpage,
+          :tags,
+          :user,
+          offline_caches: [
+            :error_messages
+          ]
+        )
+    end
 
     respond_to do |format|
       format.html { render :index }
-      format.json { render :index, status: :ok }
+      if @searcher.errors.any?
+        format.json { render :error, status: 500 }
```

Comment:
should this really be a 500?

Reply:
Hmm, probably not 🤷‍♀️ I need to figure out a better way to surface errors with the built-in [semantic-ui search component](https://semantic-ui.com/modules/search.html) though